catalyst-gateway/bin/src/state/mod.rs

```diff
@@ -49,8 +50,50 @@ impl State {
         Ok(state)
     }
 
-    // pub(crate) async fn event_db(&self) -> Option<Arc<dyn EventDbQueries>> {
-    //
-    //
-    // }
+    /// Get the reference to the database connection pool for `EventDB`.
+    pub(crate) fn event_db(&self) -> Option<Arc<dyn EventDbQueries>> {
```

Comment:
Maybe make sense change return type of `event_db()` function to `Result<Arc<dyn EventDbQueries>, Error>` instead of `Option<Db>` so if every place we don't need to make such check and just get `event_db` instance like that
```
let event_db = pool.event_db()?;
```

Reply:
Yeah, I agree. I'll change it to `Result`.